lib/realtime_web/controllers/tenant_controller.ex

```diff
@@ -244,6 +244,31 @@ defmodule RealtimeWeb.TenantController do
     end
   end
 
+  operation(:disconnect_clients,
+    summary: "Disconnect all clients",
+    parameters: [
+      token: [
+        in: :header,
+        name: "Authorization",
+        schema: %OpenApiSpex.Schema{type: :string},
+        required: true,
+        example:
+          "Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2ODAxNjIxNTR9.U9orU6YYqXAtpF8uAiw6MS553tm4XxRzxOhz2IwDhpY"
+      ],
+      tenant_id: [in: :path, description: "Tenant ID", type: :string]
+    ],
+    responses: %{
+      200 => TenantHealthResponse.response(),
+      403 => EmptyResponse.response()
+    }
+  )
+
+  def disconnect_clients(%{assigns: %{tenant: tenant}} = conn, %{"tenant_id" => tenant_id}) do
+    Tenants.disconnect_clients(tenant_id)
+    PostgresCdc.stop_all(tenant, @stop_timeout)
```

Comment:
@filipecabaco are there any potential race conditions here?

For example, what would happen if a postgres changes subscribed client gets disconnected and immediately reconnects? This client has a record in `realtime.subscriptions` again upon reconnection. Then when the call to stop PostgresCdc finally goes through would it wipe away the record? Does calling `PostgresCdc.stop_all/2` kick out clients too?

Would it be better to first call `PostgresCdc.stop_all/2` and if it returns `:ok` then call `Tenants.disconnect_clients/1`?

Reply:
This is a tricky one as we have no way of doing this without being fully async since we will need to kill potentially thounsands of processes in their own lifecycle.

We can actually suspend the user before hand, wait a couple of seconds (which will already kill socket connections) and then kill postgres cdc

Reply:
Yeah probably need to:
- suspend
- bust tenant cache
- stop_all (yeah before disconnect)
- disconnect
- unsuspend

Note: need to bust the tenant cache tho too because suspend flag is there

Reply:
If you disconnect and they can't reconnect, then yeah you could stop_all after.

Reply:
Looks like suspend busts the tenant cache already 🥳